boot/freeldr/freeldr/disk/disk.c

```diff
@@ -104,6 +104,8 @@ BOOLEAN DiskGetBootPath(OUT PCHAR BootPath, IN ULONG Size)
     if (*FrldrBootPath)
         goto Done;
 
+    BootPath[0] = ANSI_NULL;
```

Comment:
It looks to me that if you get uninitialized BootPath (with garbage) it's because also this function returned FALSE but its callers haven't checked for any failure and continued (this would be buggy)?
Unless it returned TRUE but strncpy(BootPath, FrldrBootPath, Size); misbehaved somehow ? Could you analyse this a bit further and tell me what is going on in these situations?
(Because if so, the following functions would be affected as well:
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/arm/macharm.c;hb=6ee7cba77b3a0e2f3383d4ae8dddcc52a1a55965#l99
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/freeldr/arch/i386/pcdisk.c;hb=6ee7cba77b3a0e2f3383d4ae8dddcc52a1a55965#l578
)

Reply:
Yes, it returns FALSE in case DiskGetActivePartitionEntry fails.

Reply:
And that's how you get these uninitialized paths when booting with the XBOX? But if so, why does the call fail?

Reply:
Yes, but it happens only when I'm testing boot from FATX disk, it fails because this partitioning scheme is not compatible with MBR. I will create another PR which will add abstractions for different partition types, once this one is merged. :wink: 

Also I noticed fun fact - there are no checks for returning value at all according to grep :sweat_smile: https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=MachDiskGetBootPath

Reply:
I see... :`(

Reply:
> I will create another PR which will add abstractions for different partition types, once this one is merged.

:+1: we could then have GPT support abstracted later :D

Reply:
Of course, my pending code already have GPT detection and stubs. :grin: